reporter/attrmgr.go

```diff
@@ -0,0 +1,60 @@
+package reporter // import "go.opentelemetry.io/ebpf-profiler/reporter"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/otel/attribute"
+	common "go.opentelemetry.io/proto/otlp/common/v1"
+)
+
+// AttrIndex is an index in the `Profile.attribute_table`.
+type AttrIndex = uint64
+
+// AttrTableManager maintains index allocation and deduplication for attribute tables.
+type AttrTableManager struct {
+	// indices maps compound keys to the indices in the attribute table.
+	indices map[string]AttrIndex
+
+	// attrTable being populated.
+	attrTable *[]*common.KeyValue
+}
+
+func NewAttrTableManager(attrTable *[]*common.KeyValue) *AttrTableManager {
```

Comment:
Alternatively we could not take that any argument, and rename the type to `AttrTableBuilder`. In that case I'd then add a new method `GetAttrTable()` that returns the final built table. That'd change the workflow for the caller to:

- Create empty instance
- Populate it with `AddXXX` calls
- Take the final table with `GetAttrTable()` and manually assign it to the attribute table field

Perhaps more obvious than mutating the slice field in place through a pointer? No strong preference here.

Reply:
I thought about this as well (named it `Get()` to not repeat AttrTable), but it comes with the downside that you need to call this function when done. And this can be forgotten, while the current solution is safer from accidental failure.

My not very strong preference would be to avoid an extra `Get` function call.